exchanges/sessions.py

```diff
@@ -0,0 +1,372 @@
+    """
+    Extracts irregular partial trading days from the data.
+
+    Args:
+        data (dict): The data containing irregular partial trading days.
+
+    Returns:
+        list: A list of irregular partial trading days.
+    """
+    dates_ = []
+    tzinfo = pytz.timezone(data["timezone"])
+    for yyyymmdd in data["irregular_partial_trading_days"]:
+        dates_.append(
+            tzinfo.localize(
+                datetime.strptime(yyyymmdd.strftime("%Y-%m-%d"), "%Y-%m-%d")
+            )
+        )
+    return dates_
+
+
+def get_pause_days(data):
+    """
+    Extracts pause days from the data.
+
+    Args:
+        data (dict): The data containing pause days.
+
+    Returns:
+        list: A list of pause days.
+    """
+    dates_ = []
+    tzinfo = pytz.timezone(data["timezone"])
+    for yyyymmdd in data["intra_day_interruptions"]:
+        dates_.append(
+            tzinfo.localize(
+                datetime.strptime(yyyymmdd.strftime("%Y-%m-%d"), "%Y-%m-%d")
+            )
+        )
+    return dates_
+
+
+def GetActiveSession(unix_time, filepath):
+    """
+    Gets the active trading session for a given Unix timestamp.
+    The function calls GetSessions function and get the list of all sessions for a
+    date corresponding to unix_time.
+
+    The function then chooses one session which contains the unix_time.
+
+    Args:
+        unix_time (int): The Unix timestamp to get the session for.
+        filepath (str): The path to the YAML file containing the data.
+
+    Returns:
+        dict: A dictionary representing the active trading session.
+    
+    Example:
+        >>>GetSessions(pd.Timestamp("2000-02-16", tz="America/New_York"), "nyse.yml")
+        {   'date': '2000-02-16',
+            'gte': {'status': 'PAUSE', 'time': '12:00 pm'},
+            'lt': {'status': 'REOPEN', 'time': '12:01 pm'},
+            'session': '',
+            'timezone': 'America/New_York'}
+    """
+    date_ = pd.Timestamp(unix_time, unit="ns", tz=pytz.utc).astimezone(
+        "America/New_York"
+    )
+    logging.info(f"finding a session for the time stamp {date_}")
+    sessions = GetSessions(date_, filepath)
+
+    for session in sessions:
+        tzinfo = pytz.timezone(session["timezone"])
+        session_start = tzinfo.localize(
+            datetime.strptime(
+                f"{session['date']} {session['gte']['time']}", "%Y-%m-%d %I:%M %p"
+            )
+        )
+        session_end = tzinfo.localize(
+            datetime.strptime(
+                f"{session['date']} {session['lt']['time']}", "%Y-%m-%d %I:%M %p"
+            )
+        )
+
+        if session_start <= date_ <= session_end:
+            return session
+    return {}
+
+
+def GetSessions(date_: datetime, filepath: str):
+    """
+    Gets the trading sessions for a given date.
+    The function works by following the below logic.
+
+    It checks that date_ lies in data_provided_from_date, data_provided_through_date
```

Comment:
If date falls outside this range, assume standard days/sessions (see comment on pull request)

Reply:
have modified the function. If date falls outside this range then the function will log that but will continue to construct standard sessions